project-docs/stumptown-writers-guide.md

```diff
@@ -0,0 +1,697 @@
+
+## Meta ingredient details
+
+### title
+
+Mandatory for all docs pages, even guide pages and landing pages. Not included in the recipe.
+A string that's used as the H1 heading for the doc page itself, and may also be used in other contexts such as generated lists of links.
+
+### short_title
+
+Optional for all docs pages, even guide pages and landing pages. Not included in the recipe.
+A string that's used as an alternative to `title` in certain contexts. For example the renderer might choose to display link text in sidebars using `short_title` rather than `title`.
+
+### mdn_url
+
+Mandatory for all docs pages, even guide pages and landing pages. Not included in the recipe.
+
+A string that's used by the renderer to set the URL for the docs page to build. This must be given as a relative URL, and includes the locale: `/en-US/docs/Web/HTML/Element/a`.
+
+### interactive_example
+
+An object used to specify an interactive example to include in the page. The object contains two properties:
+
+-   `url`: URL for the build interactive example, e.g. `https://interactive-examples.mdn.mozilla.net/pages/tabbed/abbr.html`
+-   `height`: String determining which height to select for the iframe. This may be any of:
+    -   "js-short"
+    -   "js-standard"
+    -   "js-tall"
+    -   "css-standard"
+    -   "html-short"
+    -   "html-standard"
+    -   "html-tall"
+
+### attributes
+
+An object that's used to describe which HTML attributes the item supports.
+
+The `attributes` ingredient contains two properties:
+
+-   `global`: Mandatory. This is a path to a directory containing the global attributes.
+-   `element-specific`: Optional. This may be either:
+    -   a path to a directory containing the element-specific attributes.
+    -   an array of file paths, each of which contains documentation for a single attribute.
+
+For example:
+
+```yaml
+attributes:
+    element_specific: ./attributes
+    global: /content/html/global_attributes
+```
+
+In this example, there should be an "attributes" subdirectory of the directory containing this front matter, and it contains files describing the element-specific attributes.
+
+```yaml
+attributes:
+    element_specific:
+        - ../input/attributes/autocomplete.md
+        - ../input/attributes/name.md
+    global: /content/html/global_attributes
+```
+
+In this example, there are two element-specific attributes, found at "../input/attributes/autocomplete.md" and "../input/attributes/name.md", relative to the directory containing this front matter.
+
+Paths may be given as an absolute path from the stumptown root directory, or a path relative to the file containing the ingredient.
+
+**Note: I think we should remove `global`, since everyone always has it and it always has the same value.**
+
+Each element-specific attribute is described in its own Markdown file, with its own syntax.
+
+#### Element-specific HTML attribute syntax
+
+Here's an example HTML attribute file:
```

Comment:
Personally, I don't like that much how attribute md files are different from say the main element md file, but this is where we're at right now. Can you maybe describe a bit how these md files are different and not really "pages". It would maybe also help me to better understand why we made it this way.

Reply:
This part of stumptown hasn't changed since Portland. Back then we designed the format of attributes files as an attempt to balance between something that could express a structure ("Type", "Values" etc) with something that would be reasonably nice to write and to read.

One of the things about attributes files is that you have a lot of structure (e.g. "Values"), with bits of pure prose attached to bits of the structure (e.g. a description of an individual value).

There are general alternative approaches, like:
* express the structure as a YAML object, and embed the prose descriptions as attributes of the object. But I think we don't really want to embed blobs of Markdown inside YAML objects (or maybe we do?). It tends to push writing towards form-filling.
* express the structure as a YAML object, and point to the prose descriptions as separate Markdown files. But this will give us very fragmentary docs.
* ?

I'm very happy to consider other options, but not sure what they are. I also think this will get interesting when we start to deal with other types of pages that have similar features (e.g. functions, which have a very clear structure (params, returns, exceptions) with bits of pure prose attached.

Reply:
I've filed https://github.com/mdn/stumptown-content/issues/215 for this.